src/vcard/mod_vcard_api.erl

```diff
@@ -0,0 +1,277 @@
+                end, [], Value);
+transform_subfield_and_value(Name, Value) ->
+    construct_xmlel(from_gql_to_xml(Name), [{xmlcdata, Value}]).
+
+proccess_child_map(Value) ->
+    io:format("~p", [maps:to_list(Value)]),
+    lists:foldl(fun({Name, SubfieldValue}, Acc) ->
+                    Acc ++ transform_subfield_and_value(Name, SubfieldValue)
+                end, [], maps:to_list(Value)).
+
+from_gql_to_xml(<<"formattedName">>) -> <<"FN">>;
+from_gql_to_xml(<<"nameComponents">>) -> <<"N">>;
+from_gql_to_xml(<<"nickname">>) -> <<"NICKNAME">>;
+from_gql_to_xml(<<"photo">>) -> <<"PHOTO">>;
+from_gql_to_xml(<<"birthday">>) -> <<"BDAY">>;
+from_gql_to_xml(<<"address">>) -> <<"ADR">>;
+from_gql_to_xml(<<"label">>) -> <<"LABEL">>;
+from_gql_to_xml(<<"telephone">>) -> <<"TEL">>;
+from_gql_to_xml(<<"email">>) -> <<"EMAIL">>;
+from_gql_to_xml(<<"jabberId">>) -> <<"JABBERID">>;
+from_gql_to_xml(<<"mailer">>) -> <<"MAILER">>;
+from_gql_to_xml(<<"timeZone">>) -> <<"TZ">>;
+from_gql_to_xml(<<"geo">>) -> <<"GEO">>;
+from_gql_to_xml(<<"title">>) -> <<"TITLE">>;
+from_gql_to_xml(<<"role">>) -> <<"ROLE">>;
+from_gql_to_xml(<<"logo">>) -> <<"LOGO">>;
+from_gql_to_xml(<<"agent">>) -> <<"AGENT">>;
+from_gql_to_xml(<<"org">>) -> <<"ORG">>;
+from_gql_to_xml(<<"categories">>) -> <<"CATEGORIES">>;
+from_gql_to_xml(<<"note">>) -> <<"NOTE">>;
+from_gql_to_xml(<<"prodId">>) -> <<"PRODID">>;
+from_gql_to_xml(<<"rev">>) -> <<"REV">>;
+from_gql_to_xml(<<"sortString">>) -> <<"SOR">>;
+from_gql_to_xml(<<"sound">>) -> <<"SOUND">>;
+from_gql_to_xml(<<"uid">>) -> <<"UID">>;
+from_gql_to_xml(<<"url">>) -> <<"URL">>;
+from_gql_to_xml(<<"desc">>) -> <<"DESC">>;
+from_gql_to_xml(<<"class">>) -> <<"CLASS">>;
+from_gql_to_xml(<<"key">>) -> <<"KEY">>;
+from_gql_to_xml(<<"lat">>) -> <<"LAT">>;
+from_gql_to_xml(<<"lon">>) -> <<"LON">>;
+from_gql_to_xml(<<"orgname">>) -> <<"ORGNAME">>;
+from_gql_to_xml(<<"locality">>) -> <<"LOCALITY">>;
+from_gql_to_xml(<<"orgunit">>) -> <<"ORGUNIT">>;
+from_gql_to_xml(<<"givenName">>) -> <<"GIVEN">>;
+from_gql_to_xml(<<"middleName">>) -> <<"MIDDLE">>;
+from_gql_to_xml(<<"family">>) -> <<"FAMILY">>;
+from_gql_to_xml(<<"prefix">>) -> <<"PREFIX">>;
+from_gql_to_xml(<<"sufix">>) -> <<"SUFIX">>;
+from_gql_to_xml(<<"pobox">>) -> <<"POBOX">>;
+from_gql_to_xml(<<"extadd">>) -> <<"EXTADD">>;
+from_gql_to_xml(<<"street">>) -> <<"STREET">>;
+from_gql_to_xml(<<"region">>) -> <<"REGION">>;
+from_gql_to_xml(<<"pcode">>) -> <<"PCODE">>;
+from_gql_to_xml(<<"number">>) -> <<"NUMBER">>;
+from_gql_to_xml(<<"line">>) -> <<"LINE">>;
+from_gql_to_xml(<<"userId">>) -> <<"USERID">>;
+from_gql_to_xml(<<"credential">>) -> <<"CRED">>;
+from_gql_to_xml(<<"keyword">>) -> <<"KEYWORD">>;
+from_gql_to_xml(<<"country">>) -> <<"CTRY">>.
+
+to_gql_format(Vcard) ->
+    lists:foldl(fun(#xmlel{name = Name, children = Value}, Acc) ->
+        maps:merge(Acc, transform_from_xml(Name, Value, Acc))
+    end, #{}, Vcard).
+
+transform_from_xml(<<"FN">>, [{_, Value}], _) ->
```

Comment:
I wonder if the conversion could be done in a more automated way similar to the from GraphQL to XML. But I think this explicit version is also OK. :)

Reply:
We could just use the original names (not the prettiest ones, I admit) and hence avoid the translation. But since we already went ahead with these nicer names, manual conversion seems to be the way to go. One could use a nested structure for specifying the format, but that would be a bit overcomplicated to me.